### PR TITLE
Update Configuration doc to use double underscores instead of colons for hierarchical config keys (4968)

### PR DESCRIPTION
**Pull request by isaac2004**
Based on conversation in #4968, changing language for hierarchigal config keys

@scottaddie @rachelappel 

[Internal Review Page](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?branch=pr-en-us-5876&tabs=basicconfiguration#configuration-by-environment)

### CODE REVIEW

File: aspnetcore/fundamentals/configuration/index.md [hunk 1]
```diff
@@ -108,8 +108,7 @@ Configuration considerations:
 * `IOptionsSnapshot` can reload configuration data when it changes. For more information, see [IOptionsSnapshot](xref:fundamentals/configuration/options#reload-configuration-data-with-ioptionssnapshot).,
 * Configuration keys are **not** case-sensitive.
 * **Never** store passwords or other sensitive data in configuration provider code or in plain text configuration files. Don't use production secrets in development or test environments. Specify secrets outside of the project so that they can't be accidentally committed to a source code repository. Learn more about [how to work with multiple environments](xref:fundamentals/environments) and managing [safe storage of app secrets in development](xref:security/app-secrets).
-* If a colon (`:`) can't be used in environment variables on a system, replace the colon (`:`) with a double-underscore (`__`).
-
+* A colon (`:`) may also be used for hierarchigal config keys, but may not work on all platforms. Double underscore (__) is supported by all platforms.
```
Comment (scottaddie) [inline aspnetcore/fundamentals/configuration/index.md]: - may also --> may
- hierarchigal --> hierarchical

Comment (isaac2004) [reply aspnetcore/fundamentals/configuration/index.md]: done

File: aspnetcore/fundamentals/configuration/index.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ The app reads and displays the following configuration settings:
 
 [!code-json[](index/sample/ConfigJson/appsettings.json)]
 
-Configuration consists of a hierarchical list of name-value pairs in which the nodes are separated by a colon. To retrieve a value, access the `Configuration` indexer with the corresponding item's key:
+Configuration consists of a hierarchical list of name-value pairs in which the nodes are separated by a double underscore (`__`). To retrieve a value, access the `Configuration` indexer with the corresponding item's key:
```
Comment (danroth27) [inline aspnetcore/fundamentals/configuration/index.md]: This isn't quite right. The double underscore syntax only applies to config values specified in env vars. When interacting with the config api you should use colons.

File: aspnetcore/fundamentals/configuration/index.md [hunk 3]
```diff
@@ -108,8 +108,7 @@ Configuration considerations:
 * `IOptionsSnapshot` can reload configuration data when it changes. For more information, see [IOptionsSnapshot](xref:fundamentals/configuration/options#reload-configuration-data-with-ioptionssnapshot).,
 * Configuration keys are **not** case-sensitive.
 * **Never** store passwords or other sensitive data in configuration provider code or in plain text configuration files. Don't use production secrets in development or test environments. Specify secrets outside of the project so that they can't be accidentally committed to a source code repository. Learn more about [how to work with multiple environments](xref:fundamentals/environments) and managing [safe storage of app secrets in development](xref:security/app-secrets).
-* If a colon (`:`) can't be used in environment variables on a system, replace the colon (`:`) with a double-underscore (`__`).
-
+* A colon (`:`) may be used for hierarchical config keys, but may not work on all platforms. Double underscore (__) is supported by all platforms.
```
Comment (danroth27) [inline aspnetcore/fundamentals/configuration/index.md]: This is only true for specifying config values in env vars.

**Comment by isaac2004** [comment]
@danroth27 how does that look?

**Comment by danroth27** [comment]
It doesn't look like my first comment in the JSON config section was addressed.

**Comment by isaac2004** [comment]
@danroth27, look better?

**Pull request merged**